internal/api/v1/service/show.go

```diff
@@ -40,19 +40,25 @@ func (sc Controller) Show(c *gin.Context) apierror.APIErrors {
 		}
 	}
 
+	appsOf, err := servicesToApps(ctx, cluster, namespace)
```

Comment:
There are various ways to improve performance and complexity here. I think this method shouldn't list all apps like it does now. It doesn't need the fully populated applications (which is fetches by calling `application.List`). It only needs the app to service binding information. This information can be extracted by simply fetching all the "binding" secrets (those created like this: https://github.com/epinio/epinio/blob/main/internal/application/services.go#L154-L159).

Fetching just those secrets should be cheaper than fetching all apps (fully populated).

Reply:
Consumers of servicesToApps seem to only need the app names, should be ok to not having the full app objects.

Reply:
from what I see, we would need some additional label on the `*-svc` secrets to filter them from `*-env` and `*-scale` ones.

Reply:
I will have a look. The `servicesToApps` as is is certainly becoming more of a bottle neck, with its extended use. And the fact that fetching an app is pulling more data now than it was at the beginning.

Reply:
I am actually not sure about the `binding secrets` ... These secrets existed for catalog services, where they were separate from the service secret itself. What we have now are only `custom services`, and for these we should not have separate binding secrets at all. As these services' secrets contain the credentials directly apps can mount them as they are.

The name/part-of labels in the referenced code look to be wrong.

I will definitely check in this area now.

That said ... Even if we do not have binding secrets at the moment, adding them to the system might be a good idea, as they are indeed make it possible to query the app/service binding in any direction without having to look into a specific (app -> service) or all applications (service -> app). You could directly select for the desired set (all for a service, all for an app, all  - last for index endpoints to use one call for everything and scatter the info, instead one per app or service).

Reply:
Aha ... The referenced code is about the application-associated secret containing the names of desired services of that app ... Which is indeed the bound services of the app, outside of the app pod ... 

So, yes, your (@jimmykarily ) idea of simply reading all these has very much merit. That said, I was correct in saying that these are not the binding secrets as we had for catalog services.

That was a good catch. These secrets sort of floated in my mind and never connected that they contain the same binding information we pull(ed) from the app pods.

Reply:
Additional label for filtering ... Likely yes. Will check if there is a common labels we can use, beyond what is already present.

Reply:
commit 04467923e1d9550b60517a9f998f673c01fbf356